mp/src/game/client/momentum/ui/leaderboards/ClientTimesDisplay.cpp

```diff
@@ -201,6 +201,8 @@ void CClientTimesDisplay::FireGameEvent(IGameEvent *event)
     if (FStrEq(type, "replay_save") && event->GetBool("save"))
```

Comment:
Hmm. Should it reload *every* replay if you only made one new one?

I believe the replay path is passed along here. Could be very possible to async load just that one and add it? Can use the same paths mostly but would save on some threading and compute.

Reply:
I say this because `hud_mapfinished` and the run uploader also load this same replay, but non-asynchronously. A future branch could be using this code you'd add here for them.

Reply:
That would make sense however I'm not really sure how this would tie in to your proposal around the top where the AsyncLoadMultiple method would only get the map name. If it was a wildcard that would work but the callback structure would need to be changed a bit.

Following this logic, OnConfirmDeleteReplay should also maybe not reload everything and only remove the entry after deleting the file. But on the other hand going through all CMomReplayBase* objects and somehow associating them to the deleted file might not be worth the hassle.

Reply:
Right -- it would be another function like `LoadReplayByFileAsync` and your current one would become `LoadReplaysForMapAsync`. Both of these should call into different callbacks but would ultimately do roughly the same thing of fetching the file via handle, handling that file (inserting and converting) then clearing the request.

I agree that OnConfirmDeleteReplay should only loop through currently loaded ones and delete it from the list and re-sort. It was also pretty naively implemented by just reloading everything.

Something to note is that those message functions can very easily be`MESSAGE_FUNC_PARAMS` that take the keyvalues in, which can hold pointers tied to a string key.